example_project/mappers/residential/samples/util.rb

```diff
@@ -3,13 +3,237 @@
 # See also https://github.com/urbanopt/urbanopt-example-geojson-project/blob/develop/LICENSE.md
 # *********************************************************************************
 
-def get_resstock_building_id(args, buildstock_csv_path)
-  '''Return ResStock building_id based on geojson file.'''
+require 'csv'
+require 'securerandom'
 
-  # TODO
+### Mapping methods from UO feature input names to buildstock characteristics names
 
-  resstock_building_id = 15 # FIXME: remove this line once this method actually assigns resstock_building_id
-  return resstock_building_id
+def get_resstock_building_id(buildstock_csv_path, feature, building_type, logger)
+  '''develop get_resstock_building_id method'''
+
+  number_of_residential_units = 1
+  begin
+    number_of_residential_units = feature.number_of_residential_units
+  rescue StandardError
+  end
+
+  mapped_properties = {}
+
+  # Required properties to do the search
+  begin
+    mapped_properties['Geometry Building Type RECS'] = map_to_resstock_building_type(building_type, number_of_residential_units)
+  rescue StandardError
+  end
+
+  # Conditionally add properties if they exist
+  # number_of_stories_above_ground
+  begin
+    mapped_properties['Geometry Stories'] = feature.number_of_stories_above_ground
+  rescue StandardError
+    logger.info("\nnumber_of_stories_above_ground for feature #{feature.id} was not used to filter buildstok csv since it does not exist for this feature")
+  end
+
+  # number_of_residential_units SFA/MF
+  begin
+    mapped_properties['Geometry Building Number Units SFA'], mapped_properties['Geometry Building Number Units MF'] = map_to_resstock_num_units(building_type, number_of_residential_units)
+  rescue StandardError
+    logger.info("\nMF number_of_residential_units for feature #{feature.id} was not used to filter buildstock csv since it does not exist for this feature")
+  end
+
+  # year built
+  begin
+   mapped_properties['Vintage ACS'] = map_to_resstock_vintage(feature.year_built) # Assuming direct mapping or apply conversion if needed
+  rescue StandardError
+   logger.info("\nyear_buit for feature #{feature.id} was not used to filter buildstok csv since it does not exist for this feature")
+  end
+
+  # floor_area
+  begin
+    mapped_properties['Geometry Floor Area'] = map_to_resstock_floor_area(feature.floor_area, number_of_residential_units)
+  rescue StandardError
+    logger.info("\nfloor_area for feature #{feature.id} was not used to filter buildstok csv since it does not exist for this feature")
+  end
+
+  # number_of_bedrooms
+  begin
+    mapped_properties['Bedrooms'] = feature.number_of_bedrooms / number_of_residential_units # Assuming direct mapping or apply conversion if needed
```

Comment:
Feature bedrooms describes the whole building, not just the unit bedrooms like ResStock does.